### ISSUE TITLE
Create a Makefile to make Coda compile on Linux

### ISSUE THREAD

**Issue by SujalChoudhari** [SujalChoudhari/Coda#28]
Coda is currently, only windows-based language.
To venture out in the world of cross-platform code, Coda needs to be set up for a build tool.

Things to implement:
running `make` command in the project directory should:
- Compile each and every file in dir and sub dirs into a single file.
- The Output file should be placed in /bin/{Configuration=Release}/{Platform=Linux} 
- It should be able to compile some files (Package files) into the dynamic link libraries.

Basically making Coda Cross-Platform.
